common/models/user.js

```diff
@@ -301,6 +301,25 @@ module.exports = function(User) {
     return fn.promise;
   };
 
+  User.observe('before delete', function(ctx, next) {
+    var AccessToken = ctx.Model.app.models.AccessToken;
+
+    AccessToken.find({
+      where: {
+        userId: ctx.where.id,
+      },
+    }, function(err, tokens) {
+      tokens.forEach(function(token) {
+        AccessToken.destroyById(token.id, function() {
```

Comment:
This code (`tokens.forEach` + `destroyById`) deletes tokens in background. It means that you would delete the user even if some of they tokens were not removed.

Also calling `destroyById` repeatedly is low performant. The best solution is to call `ctx.Model.accessTokens.destroyAll` as I commented above. For future reference, if you run into situation when that's not possible, then the second best option is to use [inq](https://docs.strongloop.com/display/LB/Where+filter#Wherefilter-inq) operator and delete multiple records at once, e.g. `AccessToken.deleteAll({ id: { inq: tokenIds } }, cb)`